src/Config/Validator/Deploy/DeprecatedSearchEngine.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Magento\MagentoCloud\Config\SearchEngine;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+
+/**
+ * Validates if a deprecated MySQL search engine is used.
+ */
+class DeprecatedSearchEngine implements ValidatorInterface
+{
+    /**
+     * @var Validator\ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var SearchEngine
+     */
+    private $searchEngine;
+
+    /**
+     * @param Validator\ResultFactory $resultFactory
+     * @param SearchEngine $searchEngine
+     */
+    public function __construct(Validator\ResultFactory $resultFactory, SearchEngine $searchEngine)
+    {
+        $this->resultFactory = $resultFactory;
+        $this->searchEngine = $searchEngine;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        if (SearchEngine::ENGINE_MYSQL === $this->searchEngine->getName()) {
+            return $this->resultFactory->error(
+                '"MySQL" is configured as a search engine. ' . PHP_EOL .
+                'This option is deprecated and will be removed in one of the next versions.'
```

Comment:
@meker12 could you please review this message?

Reply:
Maybe just say - 
The MySQL search configuration option is deprecated. Use Elasticsearch instead. See https://devdocs.magento.com for further information.

I'm quite sure any developer will figure out in 10s what to search for.....

Reply:
Since ECE-Tools can also be used without Magento Cloud (Docker, on-prem, etc) I will remove link completely.

Reply:
But https://devdocs.magento.com is applicable to all. However, if you want to leave it clean of links to docs then I'm okay with that. A developer should be able to work things out from a simple message